torch/jit/_monkeytype_config.py

```diff
@@ -16,6 +15,25 @@
 except ImportError:
     _IS_MONKEYTYPE_INSTALLED = False
 
+def get_type(type):
+    """
+    Helper function which converts the given type to a torchScript acceptable format.
+    """
+    if inspect.getmodule(type) == typing:
+        # If the type is a type imported from typing
+        # like Tuple, List, Dict then replace `typing.`
+        # with a null string. This needs to be done since
+        # typing.List is not accepted by TorchScript.
+        type_to_string = str(type)
+        return type_to_string.replace(type.__module__ + '.', ' ') + ','
+    elif 'torch' in type.__module__:
+        # If the type is a subtype of torch module, then TorchScript expects a fully qualified name
+        # for the type which is obtained by combining the module name and type name.
+        return type.__module__ + '.' + type.__name__ + ','
+    else:
+        # For all other types use the name for the type.
+        return type.__name__ + ','
+
 def get_optional_of_element_type(types: str):
```

Comment:
If we pass a list of type objects here instead of concatenated, the processing here would be easier and more robust.

Reply:
Yeah agreed. But this will go away soon , when Union gets landed. With the current ETA it's about a week or two. So, I would lean towards not cleaning this up and rather have the Union take care of it.

Reply:
Yes, it will go away soon, but it still seems dangerous and inappropriate to have fragile code in production for a couple weeks.

I just re-reviewed this function, found a few cases that I think will break current implementation:
```
print(get_optional_of_element_type("NoneType,MySpecialNoneType"))
print(get_optional_of_element_type("List[int],NoneType"))
print(get_optional_of_element_type("Dict[int, str],NoneType"))
```

I am expecting results to be
```
Optional[MySpecialNoneType]
Optional[List[int]]
Optional[Dict[int, str]]
```

While actual outputs are:
```
Optional[NoneType]
List[Optional[int]]
Optional[ str]]
```

Some of these bugs are due to hard-coded magic strings, some of them are due to incorrectly splitting types str by comma without verifying there are only two parts from the split.

IMHO, it is worthwhile to implement a more robust implementation that operate on type objects rather than strings before `Union` lands. Let me know what you think.